src/prefect/tasks/jupyter/jupyter.py

```diff
@@ -19,7 +19,10 @@ class ExecuteNotebook(Task):
         - parameters (dict, optional): dictionary of parameters to use for the notebook
             Can also be provided at runtime
         - output_format (str, optional): Notebook output format.
+            This argument is gnored if `exporter` is provided.
             Currently supported: json, html (default: json)
+        - exporter (nbconvert.Exporter, optional): The exporter that is used for
+            exporting the notebook, regardless of `output_format`.
```

Comment:
Interesting, I had originally envisioned supporting this as a combination of `output_format` (for specifying the exporter class) and optional `exporter_kwargs` for instantiating that class. Docstring would be something like:

```python
"""
output_format (str, optional): Notebook output format. One of {'json', 'html'},
  or a full import path to a `nbconvert.Exporter` class.
exporter_kwargs (dict, optional): Extra arguments to use when instantiating
  the `nbconvert.Exporter`.
"""
```

I think this is nice, since many users won't need to configure the exporters, so using a simple name alias (like "html") is sufficient. And for those that need to configure extra options, they can pass them in as a simple dict rather than having to import from `nbconvert`.

I saw that @madkinsz suggested the approach you're using, but I don't find that argument convincing - really the `json` output format should be passing through a `nbconvert.exporters.notebook.NotebookExporter` anyway.

Could go either way on this though, just my 2 cents.

Reply:
Yeah I don't like that the `output_format` gets ignored in this implementation. I think we should actually be using https://github.com/jupyter/nbconvert/blob/master/nbconvert/exporters/base.py#L91 to retrieve the exporter passed as `output_format` and then initialize it with `exporter_kwargs`. This would automatically support a bunch of short names as well as full import paths.

```
In [8]: get_export_names()
Out[8]: 
['asciidoc',
 'custom',
 'html',
 'latex',
 'markdown',
 'notebook',
 'pdf',
 'python',
 'rst',
 'script',
 'slides']

In [9]: get_exporter('html')
Out[9]: nbconvert.exporters.html.HTMLExporter
``` 

It's not 100% clear to me how the json format should be handled though.

Reply:
Ah, didn't know about `get_exporter`, yeah we should move to use that. For `json`, this matches `notebook` - we should change the default to `notebook` and keep around `json` as an alias.

Reply:
I agree with @madkinsz that `export` in combination with `get_exporter` is the way to go.

Two things I'm still not entirely sure about:
1. Should you still be able to provide your own Exporter instance or are short names and import paths for `get_exporter` handles enough?
2. How to document the valid output_format options, for now I've just listed all of them.

Reply:
re 2. I'd point them to the nbconvert `get_export_names()` as well but it's not a big deal.